r/tests/testthat/test-dplyr-funcs-string.R

```diff
@@ -757,9 +757,10 @@ test_that("strftime", {
 
   times <- tibble(
     datetime = c(lubridate::ymd_hms("2018-10-07 19:04:05", tz = "Etc/GMT+6"), NA),
-    date = c(as.Date("2021-09-09"), NA)
+    date = c(as.Date("2021-01-01"), NA)
   )
   formats <- "%a %A %w %d %b %B %m %y %Y %H %I %p %M %z %Z %j %U %W %x %X %% %G %V %u"
+  formats_date <- "%a %A %w %d %b %B %m %y %Y %j %U %W %x %G %V %u"
```

Comment:
Why not test all format flags available?

Reply:
for `format_date` I removed all the flags that are time-related since dates should not have any time components. At least some of the flags will error because of the lack of timezone on a date now (while the message could be better, having it error instead of returning a made up time value is a good behavior)

Reply:
Shouldn't time just be midnight for both R and arrow?
Agreed with removing `%z %Z`.